artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/cursor/impl/PageCursorProviderImpl.java

```diff
@@ -179,11 +182,13 @@ public void close(PageSubscription cursor) {
    }
 
    @Override
-   public void scheduleCleanup() {
+   public Future<Boolean> scheduleCleanup() {
+      final SimpleFutureImpl<Boolean> future = new SimpleFutureImpl<>();
       if (!cleanupEnabled || scheduledCleanup.intValue() > 2) {
          // Scheduled cleanup was already scheduled before.. never mind!
          // or we have cleanup disabled
-         return;
+         pagingStore.execute(() -> future.set(true));
```

Comment:
Is there a reason why you need to fire off a task into the executor just to return a completed future?  Seems simpler to just return CompletableFuture.completedFuture or just complete this one and return it directly.

Reply:
Yes... I wanted to make sure any previous call to cleanup was finished. Say you called scheduleCleanup a couple of times.. then you call scheduleCleanup again...


 I don't need to schedule another one as I already have a call scheduled, but I want to return a Future that will be released after these calls have finished.

Reply:
I am amending the comment before future.set call as:

         // Scheduled cleanup was already scheduled before.
         // On that case just flush the executor returning the future.set(true) 
         // after any previous scheduled cleanup is finished.

Reply:
marking it as resolved.. please unresolved it if you find any issues with my comments please?